manifests/init.pp

```diff
@@ -36,60 +36,48 @@
 #   }
 #
 class python (
-  Enum['absent', 'present', 'latest'] $ensure     = $python::params::ensure,
-  $version                                        = $python::params::version,
-  Enum['absent', 'present', 'latest'] $pip        = $python::params::pip,
-  Enum['absent', 'present', 'latest'] $dev        = $python::params::dev,
-  Enum['absent', 'present', 'latest'] $virtualenv = $python::params::virtualenv,
-  Enum['absent', 'present', 'latest'] $gunicorn   = $python::params::gunicorn,
-  Boolean $manage_gunicorn                        = $python::params::manage_gunicorn,
-  Boolean $manage_python_package                  = $python::params::manage_python_package,
-  Boolean $manage_virtualenv_package              = $python::params::manage_virtualenv_package,
-  Boolean $manage_pip_package                     = $python::params::manage_pip_package,
-  $gunicorn_package_name                          = $python::params::gunicorn_package_name,
-  Optional[Enum['pip', 'scl', 'rhscl', 'anaconda', '']] $provider = $python::params::provider,
-  $valid_versions                                 = $python::params::valid_versions,
-  Hash $python_pips                               = {},
-  Hash $python_virtualenvs                        = {},
-  Hash $python_pyvenvs                            = {},
-  Hash $python_requirements                       = {},
-  Hash $python_dotfiles                           = {},
-  Boolean $use_epel                               = $python::params::use_epel,
-  $rhscl_use_public_repository                    = $python::params::rhscl_use_public_repository,
-  Stdlib::Httpurl $anaconda_installer_url         = $python::params::anaconda_installer_url,
-  Stdlib::Absolutepath $anaconda_install_path     = $python::params::anaconda_install_path,
-  Boolean $manage_scl                             = $python::params::manage_scl,
-  Optional[Pattern[/[0-7]{1,4}/]] $umask          = undef,
+  Enum['absent', 'present', 'latest'] $ensure                      = $python::params::ensure,
+  Python::Version                     $version                     = $python::params::version,
+  Enum['absent', 'present', 'latest'] $pip                         = $python::params::pip,
+  Enum['absent', 'present', 'latest'] $dev                         = $python::params::dev,
+  Enum['absent', 'present', 'latest'] $virtualenv                  = $python::params::virtualenv,
+  Enum['absent', 'present', 'latest'] $gunicorn                    = $python::params::gunicorn,
```

Comment:
how is there no stdlib type for these?

Reply:
There surely is one. But I didn't want to change too much for now.

Reply:
I have looked up, and didn't find one in stdlib. But I can create a new one with within the module